### PR TITLE
Implement View Users Page

### LINKED ISSUE #59: Create a "View Users" page
We should create a "View Users" page for the admins to use. This would be like one dashboard where they can see a table of all users, and have functionality to sort, search and filter in the table (see #51 #52 #53). We can tie this with #58 so that the users can be clicked on to go to their specific users page. 

# Acceptance Criteria

- [ ] Create a "View Users" page
- [ ] Use the table component to show a list of users
- [ ] Make users navigate to their respective URL: `/users/${userId}`
- [ ] Make table pageable
- [ ] Add search box, sortable columns

### PR DESCRIPTION
**Pull request by bchampp**
Closes #59 

image

### CODE REVIEW

File: client/src/components/global/Nav.tsx [hunk 1]
```diff
@@ -47,6 +47,8 @@ export default function HourLoggerNav() {
                   )}
                 </NavDropdown>
 
+                <Nav.Link href="/users">Users</Nav.Link>
```
Comment (bchampp) [inline client/src/components/global/Nav.tsx:50]: @avalittle wasn't sure which kinds of users should be able to see this. Let me know and ill update

File: client/src/pages/UsersPage.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+export const UsersPage = () => {
+  const [loading, setLoading] = useState(true);
+  const [usersCheckedIn, setUsersCheckedIn] = useState(0);
+  const [users, setUsers] = useState(null);
+  const [query, setQuery] = useState("");
+
+  useEffect(() => {
+      // TODO: Sort users by name alphabetically
+    loadUsers();
+  }, [query]);
```
Comment (bchampp) [inline client/src/pages/UsersPage.tsx:24]: Having [query] in the dependency array is telling react that every time the value of query changes, let's call `loadUsers` again and update the UI. The query isn't being used in the API call, but will be after #51 is done.

File: client/src/pages/UsersPage.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        </tr>
+      </thead>
+      <tbody>
+        {
+          // @ts-ignore
+          users.map((user) => (
+            <tr key={user.userId}>
+              <td><a href={`/users/${user.userId}`}>
+                {user.firstName + " " + user.lastName}
+              </a>
```
Comment (bchampp) [inline client/src/pages/UsersPage.tsx:107]: Every persons name in the table is clickable, and brings you to their respective user page (to be implemented in #58)

File: api/serverless.yml [hunk 4]
```diff
@@ -16,6 +16,7 @@ provider:
     - Effect: Allow
       Action:
         - dynamodb:GetItem
+        - dynamodb:DeleteItem
```
Comment (bchampp) [inline api/serverless.yml:19]: This got lost in the sauce

File: client/src/components/global/Nav.tsx (hunk 1 above)
Comment (avalittle) [reply client/src/components/global/Nav.tsx:50]: just admin

**Pull request merged**